src/internal/cluster/tunnel.go

```diff
@@ -367,6 +367,10 @@ func (tunnel *Tunnel) checkForZarfConnectLabel(name string) error {
 		tunnel.namespace = svc.Namespace
 		// Only support a service with a single port.
 		tunnel.remotePort = svc.Spec.Ports[0].TargetPort.IntValue()
+		// if targetPort == 0, look for Port (which is required)
+		if tunnel.remotePort == 0 && svc.Spec.Ports[0].Port != 0 {
+			tunnel.remotePort = int(svc.Spec.Ports[0].Port)
+		}
```

Comment:
Technically this will fail if the `svc.Spec.Ports[0].Port` is different from the `svc.Spec.Ports[0].TargetPort` that is defined as a `containerPort` on the `Pod`.  A lot of the time these are the same, but they don't have to be.  Instead we should probably keep the value as an `intstr.IntOrString` in the `Tunnel` struct and then decipher the real port later when we have pod information in `establish()`.

Also can we move this to a branch in `defenseunicorns/zarf` (makes it easier to build local test versions).

Reply:
resolved in https://github.com/defenseunicorns/zarf/pull/1894